File: src/main/java/parser/ComparisonTokenType.java
package parser;

public enum ComparisonTokenType implements IComparisonTokenType {
    EQEQ,
    EXCL,
    EXCLEQ,
    LT,
    LTEQ,
    GT,
    GTEQ;

    @Override
    public String toString() {
        switch (this) {
            case EQEQ:
                return "==";
            case EXCL:
                return "!";
            case EXCLEQ:
                return "!=";
            case LT:
                return "<";
            case LTEQ:
                return "<=";
            case GT:
                return ">";
            case GTEQ:
                return ">=";
        }

        return "";
    }
}

File: src/main/java/parser/ComparisonTokenType.kt
package parser

enum class ComparisonTokenType : IComparisonTokenType {
    EQEQ,
    EXCL,
    EXCLEQ,
    LT,
    LTEQ,
    GT,
    GTEQ;

    override fun toString(): String {
        return when (this) {
            EQEQ -> "=="
            EXCL -> "!"
            EXCLEQ -> "!="
            LT -> "<"
            LTEQ -> "<="
            GT -> ">"
            GTEQ -> ">="
        }
    }
}
